src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -332,9 +323,32 @@ private static void RecordRequestStartEventLog(HttpContext httpContext)
             out var requestId,
             out var traceState);
 
+        var activity = _activitySource.CreateActivity(
```

Comment:
Should parent `ActivityContext` be passed in here with `IsRemote` set as `true` instead of ParentID?. So that the sampler gets correct details. With ParentID, `IsRemote` comes in as `false`. Not sure if there is another way to achieve that. @noahfalk @cijothomas.

My test
```csharp
using System;
using System.Diagnostics;

class Program
{
    static void Main(string[] args)
    {
        ActivitySource src = new ActivitySource("Test");
        using var listener = new ActivityListener
        {
            ShouldListenTo = activitySource => true,
            Sample = (ref ActivityCreationOptions<ActivityContext> options) => ComputeActivitySamplingResult(options)
        };
        ActivitySource.AddActivityListener(listener);

        var parentId = "00-35aae61e3e99044eb5ea5007f2cd159b-40a8bd87c078cb4c-00";

        // Create using parentID
        // var _activity = src.CreateActivity("Test", ActivityKind.Server, parentId);

        var ctx = new ActivityContext(ActivityTraceId.CreateFromString("35aae61e3e99044eb5ea5007f2cd159b"),
            ActivitySpanId.CreateFromString("40a8bd87c078cb4c"),
            ActivityTraceFlags.None,
            null,
            true);

        // Create using ctx
        var _activity = src.CreateActivity("Test", ActivityKind.Server, ctx);
    }

    private static ActivitySamplingResult ComputeActivitySamplingResult(
        in ActivityCreationOptions<ActivityContext> options)
    {
        var isRemote = options.Parent.IsRemote;
        Console.WriteLine(isRemote);
        return ActivitySamplingResult.AllDataAndRecorded;
    }
}
```

Reply:
Good callout!

The extractor itself is expected to set IsRemote as true. It looks like extractor is not extracing `ActivityContext` but rather doing the string traceparent only. I think the best that can be done now, is to create ActivityContext here from the extracted traceparent, and set IsRemote to true, and pass it to the StartActivity call.

Reply:
@tarekgh 

This looks like a deeper issue. The interface for the propagator doesn't have any parameter to pass out the isRemote boolean though I would think isRemote=true is implied if you are in a scenario where the id needs to be deserialized from a message. CreateActivity() also has no parameter to pass it in even if the propagator did specify it. ActivityContext does have the isRemote property, but converting the string id to ActivityContext is only legal for W3C TraceParent format ids, not for hierarchical format ids. Internally ActivitySource.CreateActivity determines the ID format is W3C and converts it to an ActivityContext at which point it is an arbitrary decision to resolve the ambiguity isRemote = false. 

https://source.dot.net/#System.Diagnostics.DiagnosticSource/System/Diagnostics/ActivityCreationOptions.cs,60

isRemote = false is probably the rare case in practice so it is unfortunate I didn't spot it. Changing that behavior is technically breaking, but maybe we should do it. Another alternative would be adding more overloads where we can pass in more information that eliminates the isRemote ambiguity. Thoughts?

Reply:
Can the caller try to parse the Id to ActivitContext before calling CreateActivity and set the desired remote value? If the Id cannot be parsed, do we care much about this case? If we care, I think we go with the breaking change, and it is not worth introducing new APIs just for this case.

Reply:
yeah it is possible for the caller to do it as a workaround, but is that the pattern you'd want to document as the proper way for everyone to use the API going forward? Effectively the 'HelloWorld' of receiving a message and starting an Activity would change from:

```C#
using(var activity = src.CreateActivity("Test", ActivityKind.Server, id))
{
   ...
}
```

into:

```C#
Activity? a = null;
if(ActivtyContext.TryParse(id, isRemote:true, out ActivityContext ctx))
{
    activity = src.CreateActivity("Test", ActivityKind.Server, ctx);
}
else
{
    activity = src.CreateActivity("Test", ActivityKind.Server, id);
}
using(activity)
{
   ...
}
```

Reply:
```C#
            using (Activity? a = ActivityContext.TryParse("Id", "", true, out ActivityContext context) ? aSource.CreateActivity("A", ActivityKind.Server, context) : aSource.CreateActivity("A", ActivityKind.Server, "id"))
            {

            }
```
Not perfect but acceptable, I guess. Also, in general, should we promote using ActivityContext over Id's?

Reply:
> Not perfect but acceptable, I guess

You can make the call. Neither option is risk-free. Two downsides of this one I could foresee:
1. Other implementers who aren't ASP.NET are unlikely to replicate the workaround so the error will probably be repeated in the future.
2. This pattern forces paying a small perf cost to parse the ID prior to knowing if there are any listeners who care about it.

> Also, in general, should we promote using ActivityContext over Id's?

I think this still comes down to how much effort should be make for back-compat with hierarchical IDs. So far we've seen very little feedback saying people cared about hierarchical so I think its fine for us to continue moving away from it while keeping an eye out for feedback.

Reply:
@noahfalk @tarekgh What do you think? Is this worth changing or should I leave it as-is?

Reply:
Is there a way to achieve this on .NET 6? The `isRemote` option on `TryParse` seems new in 7 right? I think there was some desire to have this change backported to 6. If that's the case, I think there's value in having the behavior be consistent across versions.

Reply:
I expect aspn.net will reference the latest `System.Diagnostics.DiagnosticSource` package that will ship with .NET 7.0. no? 

If you still want to work around the issue, you can have the helper method:

```C#
        private static bool TryParseActivityContext(string traceParent, string traceState, bool isRemote, out ActivityContext context)
        {
            if (ActivityContext.TryParse(traceParent, traceState, out context))
            {
                if (isRemote)
                {
                    context = new ActivityContext(context.TraceId, context.SpanId, context.TraceFlags, context.TraceState, isRemote);
                }
                return true;
            }
            
            return false;
        }
```

and then the calling code will look like:

```C#
            using (Activity? a = aSource.HasListeners() ? 
                                 TryParseActivityContext("Id", "", true, out ActivityContext context) ?  aSource.CreateActivity("A", ActivityKind.Server, context) : aSource.CreateActivity("A", ActivityKind.Server, "id") : 
                                 null)
            {
                ...
            }
```

Reply:
Thanks @tarekgh. I just pushed an update using this approach (verified that it would apply the same way to 6.0 as well if we decide to backport it).